mono/metadata/metadata.c

```diff
@@ -2580,6 +2580,9 @@ get_image_set (MonoImage **images, int nimages)
 		set->gmethod_cache = g_hash_table_new_full (inflated_method_hash, inflated_method_equal, NULL, (GDestroyNotify)free_inflated_method);
 		set->gsignature_cache = g_hash_table_new_full (inflated_signature_hash, inflated_signature_equal, NULL, (GDestroyNotify)free_inflated_signature);
 
+		set->szarray_cache = g_hash_table_new_full (mono_aligned_addr_hash, NULL, NULL, NULL);
+		set->array_cache = g_hash_table_new_full (mono_aligned_addr_hash, NULL, NULL, NULL);
+
```

Comment:
So we assume arrays of generic instances are common and preemptively init the caches, but pointers are rare so we don't create a ptr cache until we need it.

Reply:
That is what the image does. We can do this lazily as well?

Reply:
I mean it seems reasonable - we would expects pointers to generic instances be fairly rare since they're just a weird reflection thing and not otherwise legal - I was just trying to understand why the code looked different.